addons/binding/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderLocalWatcherHandler.java

```
@@ -0,0 +1,234 @@
+                for (File file : fileList) {
+
+                    if (file.isDirectory()) {
+
+                        if (recursive) {
+                            listFiles(file.getAbsolutePath(), listHidden, recursive);
+                        }
+                    }
+                    if (file.isFile()) {
+                        // logger.debug(file.getAbsolutePath());
+                        currentLocalListing.add(file.getAbsolutePath());
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    protected Runnable periodicExecutionRunnable = new Runnable() {
+        @Override
+        public void run() {
+
+            String RootDir = config.localDir;
+            ArrayList<String> diffLocalListing = new ArrayList<String>();
+
+            /*
+             * if (ftpRootDir.endsWith("/")) {
+             * ftpRootDir = ftpRootDir.substring(0, ftpRootDir.length() - 1);
+             * }
+             *
+             * if (!ftpRootDir.startsWith("/")) {
+             * ftpRootDir = "/" + ftpRootDir;
+             * }
+             */
+            currentLocalListing.clear();
+
+            try {
+                listFiles(RootDir, config.listHiddenLocal, config.listRecursiveLocal);
+            } catch (IOException e) {
+
+                logger.debug("Can't read directory: {}", config.localDir);
+            }
+
+            try {
+                diffLocalListing = FileTools.listNewFiles(previousLocalListing, currentLocalListing,
+                        currentLocalListingFile);
+            } catch (IOException e1) {
+
+                logger.debug("Error geeting new file names");
+            }
+
+            for (String newLocalFile : diffLocalListing) {
+
+                triggerChannel(CHANNEL_LOCALFILENAME, newLocalFile);
+
+                try {
+                    Thread.sleep(3000);
+                } catch (InterruptedException e2) {
+
```

don't swallow the exception